lib/Sema/BuilderTransform.cpp

```diff
@@ -905,6 +910,720 @@ class BuilderClosureVisitor
 #undef CONTROL_FLOW_STMT
 };
 
+class ResultBuilderTransform
+    : private BuilderTransformerBase,
+      private StmtVisitor<ResultBuilderTransform, NullablePtr<Stmt>,
+                          NullablePtr<VarDecl>> {
+  friend StmtVisitor<ResultBuilderTransform, NullablePtr<Stmt>,
+                     NullablePtr<VarDecl>>;
+
+  using UnsupportedElt = SkipUnhandledConstructInResultBuilder::UnhandledNode;
+
+  /// The constraint system this transform is associated with.
+  ConstraintSystem &CS;
+  /// The result type of this result builder body.
+  Type ResultType;
+
+  /// The first recorded unsupported element discovered by the transformation.
+  UnsupportedElt FirstUnsupported;
+
+public:
+  ResultBuilderTransform(ConstraintSystem &cs, DeclContext *dc,
+                         Type builderType, Type resultTy)
+      : BuilderTransformerBase(&cs, dc, builderType), CS(cs),
+        ResultType(resultTy) {}
+
+  UnsupportedElt getUnsupportedElement() const { return FirstUnsupported; }
+
+  BraceStmt *apply(BraceStmt *braceStmt) {
+    auto newBody = visitBraceStmt(braceStmt, /*bodyVar=*/nullptr);
+    if (!newBody)
+      return nullptr;
+    return castToStmt<BraceStmt>(newBody.get());
+  }
+
+protected:
+  NullablePtr<Stmt> failTransform(UnsupportedElt unsupported) {
+    recordUnsupported(unsupported);
+    return nullptr;
+  }
+
+  Expr *buildCallIfWanted(SourceLoc loc, Identifier fnName,
```

Comment:
This is strategic :)  I have moved `buildCallIfWanted` to the base class and overrode it here so we don't to the checks in new implementation, I continued to use `buildCallIfWanted` because it would make it safer/easier to switch from it once the base class and pre-existing implementation is removed.

Reply:
Ah got it, I missed the `override` here, thanks!

Reply:
Np!